mep/accounts/models.py

```diff
@@ -350,6 +350,35 @@ def event_label(self):
                 return self.nonstandard_notation[match.group(0)]
         return self.event_type
 
+    def calculate_duration(self):
+        '''Calculate and set subscription duration based on start and end
+        date, when both are known. Duration is only calculated when both
+        dates are fully known OR if both dates are month/day only (year
+        unknown), in which case we assume dates are sequential in the
+        same or subsequent year.'''
+
+        # full precision is either all flags or null/none
+        full_precision = [DatePrecision.year | DatePrecision.month |
+                          DatePrecision.day, None]
```

Comment:
this is kinda confusing, do bitflags normally work this way?

Reply:
They do — but I just looked, and there is an `all_flags` shortcut that would probably be more readable, I'll switch to that.  (fwiw, also learned that there is built-in support for flags in the enum module starting in python 3.6).

Reply:
good to know on both of those! +1 for `all_flags` if it's easy.